### DISCUSSION TITLE
How to sync a single directory to OneDrive outside local OneDrive root directory?

### DISCUSSION THREAD

**Discussion by martonlente** [abraunegg/onedrive#2344] [Q&A] [answered]
Hello,

I'd like to (one-way) sync a single directory from within my user directory e.g. `~/test-directory` to OneDrive, without moving the directory into the default `sync_dir` (e.g. `~/OneDrive`).

Based on the documentation, I assumed the following command should work:
`onedrive --synchronize --single-directory '../test-directory' --upload-only --verbose`

However it fails, and I receive the following error messages:
```
ERROR: There was an error performing this operation on OneDrive
ERROR: Increase logging verbosity to assist determining why.
```

The client reads the local files correctly, but the errors happen because it tries to create a folder named _'../test-directory'_ in OneDrive, which is invalid. I'd like to get rid of _'../'_ part of the remote folder name.

How should I achieve the above? My client config is in its default state, and I'd like to keep it that way if possible (thinking of the above as casual sync of individual folders outside OneDrive `sync_dir`).

Thank you for your help in advance,
Márton Lente

**Comment by abraunegg** [comment, marked as answer]
@martonlente 
> How to sync a single directory to OneDrive outside local OneDrive root directory?

You cant. The folder must be present in your configured 'sync_dir' to sync to OneDrive. This is due to the OneDrive API.

**Reply by martonlente** [reply]
Hello @abraunegg,

Ok, thank you for the information!
